offer-manager-domain/src/test/groovy/life/qbic/business/products/create/CreateProductSpec.groovy

```diff
@@ -0,0 +1,90 @@
+package life.qbic.business.products.create
+
+import life.qbic.business.exceptions.DatabaseQueryException
+import life.qbic.datamodel.dtos.business.ProductId
+import life.qbic.datamodel.dtos.business.services.AtomicProduct
+import life.qbic.datamodel.dtos.business.services.Product
+import life.qbic.datamodel.dtos.business.services.ProductUnit
+import spock.lang.Specification
+
+/**
+ * <h1>Tests for the CreateProduct use case</h1>
+ *
+ * <p>This specification contains tests for all steps of the create product use case</p>
+ *
+ * @since 1.0.0
+ */
+class CreateProductSpec extends Specification {
+    CreateProductOutput output
+    ProductId productId
+    Product product
```

Comment:
```suggestion
    @Shared CreateProductOutput output
    @Shared ProductId productId
    @Shared Product product
```

Reply:
What does that do?

Reply:
https://spockframework.org/spock/javadoc/1.0/index.html?spock/lang/Shared.html

Reply:
You need to add the import manually statement though

Reply:
Aren't they automatically shared as is? I do not get it

Reply:
They are but it's just good practice I think.

Reply:
Do they brind other implications with them?

Reply:
According to https://spockframework.org/spock/docs/1.0/spock_primer.html they have the same effect as `static`. Usually fields are setup for each feature method. Adding `@Shared` prevents that. I think it is worth a try in this case

Reply:
Ah but does not work in this case. The mocks and stubs have to be set before every feature method it seems. (`setupSpec` == `@Shared`) does not work

Reply:
I would prefer to leave it as is since this works and we anyhow depend on the `setup` method here.